test/libsolidity/GasCosts.cpp

```diff
@@ -129,9 +129,9 @@ BOOST_AUTO_TEST_CASE(string_storage)
 		}
 	}
 	else if (evmVersion < EVMVersion::istanbul())
-		CHECK_DEPLOY_GAS(125829, 118559, evmVersion);
+		CHECK_DEPLOY_GAS(125829, 107949, evmVersion);
```

Comment:
Apparently, that's because they've been out of date for a while on `develop`. @matheusaaguiar updated the costs on his machine to the current ones, but turns out that CI does not even check them. We just don't run soltest on the combination of abicoder v1 and older EVM versions.

Updating them is not wrong, but it creates a misleading impression that it's the result of changes in the PR so I'd remove this. Since it's not even being checked, the CI should still pass after we do that.

Reply:
By the way, I think we might be better off converting the whole `GasCosts.cpp` into normal semantic tests, which can check costs too. It only contains two cases and I think it might have been some early approach to measuring gas that was quickly abandoned and eventually supplanted by the mechanism we have now in semantic tests. It has also snowballed into something big and messy, while the original (added in #4779) was much simpler: https://github.com/ethereum/solidity/blob/d80d324de7fa7eb79e21d1e6d7cbca4ba3eebcd0/test/libsolidity/GasCosts.cpp#L71-L82

I think we just keep adding more cases here without paying attention to the numbers and that's pointless.

I'd do that in a separate PR though. It has nothing to do with `PUSH0`.

Reply:
I updated the costs here originally because I just manually ran all the combinations of flags thinking that those would cause a CI failure. Only later, I found out that for some evm versions and flags, we don't update or check anymore.
At first, I thought that the changes were due to the addition of the new Peephole Optimizer Method and tried to confirm it. Since that was not the case, I kept digging and regressing the changes until I realized that the costs were already changing when I tried that on `develop` branch. Then after talking with @cameel , we confirmed that indeed was a problem of the tests being out-of-date for a long time.